defos/src/defos_linux.cpp

```diff
@@ -512,6 +519,24 @@ void defos_update()
 
     window_has_focus = focused_window == win;
     apply_cursor_visible();
+
+	// Cursor enter/exit events
+	if (last_cursor_event != cursor_event.type)
+	{
+		switch (cursor_event.type)
+		{
+		case LeaveNotify:
+			defos_emit_event(DEFOS_EVENT_MOUSE_LEAVE);
+			break;
+		case EnterNotify:
+			defos_emit_event(DEFOS_EVENT_MOUSE_ENTER);
+			break;
+		}
+	}
+	
+	last_cursor_event = cursor_event.type;
+	while(XPending(disp))
+		XNextEvent(disp, &cursor_event);
```

Comment:
You should do the event processing in the loop. Otherwise, you're processing only the last event.

Also, Defold might actually be eating up some events, judging by reading the GLFW 2.7 source code (Defold's windowing system is a fork of that).

https://sourceforge.net/projects/glfw/files/glfw/2.7.9/glfw-2.7.9.tar.gz/download